pkg/virt-controller/watch/vm.go

```diff
@@ -1271,14 +1272,7 @@ func (c *VMController) setupVMIFromVM(vm *virtv1.VirtualMachine) *virtv1.Virtual
 
 func (c *VMController) applyFlavorToVmi(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
 
-	flavorSpec, err := c.flavorMethods.FindFlavorSpec(vm)
-
-	if err != nil {
-		return err
-	}
-
-	preferenceSpec, err := c.flavorMethods.FindPreferenceSpec(vm)
-
+	flavorSpec, preferenceSpec, err := c.storeFlavorControllerRevisions(vm)
```

Comment:
it would be nice if we could run the `ApplyToVMI` function using a flavor and preference before the revision is written for the first time. I think it would be uncommon for any issues to occur here, but it's possible.

Reply:
ACK, I've switched this around thanks.